core/src/main/java/org/apache/calcite/rel/rules/SubQueryRemoveRule.java

```diff
@@ -946,6 +948,11 @@ private static void matchJoin(SubQueryRemoveRule rule, RelOptRuleCall call) {
       builder.push(join.getLeft());
       builder.push(join.getRight());
 
+      final RelOptUtil.Logic logic =
```

Comment:
The two logic variables in the `if/else` seem identical, in case let's move it to its original location in its updated form